### ISSUE TITLE
Select component: Item hover/focus state is not accessible

### ISSUE THREAD

**Issue by awesomephant** [SWRdata/components#12]
![image](https://github.com/user-attachments/assets/b05be5ad-d1b0-4723-a238-7e486da005bd)

See [WCAG 2.2 1.4.1](https://www.w3.org/TR/WCAG22/#use-of-color)

> Color is not used as the only visual means of conveying information, indicating an action, prompting a response, or distinguishing a visual element.

I suggest `text-decoration: underline` + `cursor: pointer`. Would also suggest getting rid of the background colour transition so it feels more responsive/is in sync with the non-transitioning underline and pointer change.
